liblumen_alloc/src/erts/term/binary/maybe_aligned_maybe_binary.rs

```diff
@@ -252,3 +254,44 @@ macro_rules! impl_maybe_aligned_try_into {
 
 impl_maybe_aligned_try_into!(MatchContext);
 impl_maybe_aligned_try_into!(SubBinary);
+
+#[derive(Debug, Error)]
+pub enum TryIntoByteVecError {
+    #[error("bitstring is not a binary")]
+    NotABinary,
+}
+
+#[derive(Debug, Error)]
+pub enum TryIntoStringError {
+    #[error("aligned, binary bitstring is not valid UTF-8")]
+    Aligned(#[from] Utf8Error),
```

Comment:
Do we ever dispatch on these two variants? I think we could just make this a single `InvalidUtf8 { is_aligned: bool, #[from] error: Utf8Error }` variant. Knowing whether or not the input was aligned is really only useful in debug output, we can't meaningfully act on it as far as I know.

Reply:
Originally I thought I had to do this because of `FromUtf8Error` vs `Utf8Error`, but then I found that you have to ask for the [`FromUtf8Error::utf8_error()`](https://doc.rust-lang.org/std/string/struct.FromUtf8Error.html#method.utf8_error), so there's not really a good reason to keep them separate.